liblangutil/Scanner.cpp

```diff
@@ -271,12 +273,74 @@ bool Scanner::skipWhitespaceExceptUnicodeLinebreak()
 	return sourcePos() != startPosition;
 }
 
+// A little helper class for counting depth of RLO/LRO/PDF uses.
+struct Scanner::UnicodeDirectionOverrideProcessor
```

Comment:
Do we actually need to do this during scanning? Would it be possible to analyze the comment string instead?